app/controllers/reactions_controller.rb

```diff
@@ -101,5 +101,12 @@ def remove_count_cache_key
     return unless params[:reactable_type] == "Article"
 
     Rails.cache.delete "count_for_reactable-Article-#{params[:reactable_id]}"
+    article = Article.find_by(id: params[:reactable_id])
+    return unless article&.public_reactions_count.presence <= 1 && Reaction.public_reaction_types.include?(params[:category])
+
+    # Bust edge if article count is 1 or zero to alter state of cache.
+    # If 1 or 0, we account for both creation and deletion of the first reaction
+    # Without knowing that at this stage yet.
+    EdgeCache::BustArticle.call(article)
```

Comment:
I wound up asking this on #19147, but it properly belongs here: How did this work before? Why weren't we busting the article cache prior to now?

Reply:
@jaw6 We are busting the cache under scenarios where the content of the post change, or a comment is added/updated.

The way reactions are fetched async, we opportunistically do not bust the article show page, in order to leave it more fresh more of the time.

But with this change, we introduce a scenario where the HTML needs to change when we go from 0 to 1 reaction, or else we would introduce content layout shift — so now we need to add a scenario where the page is busted and we send out a cold version.

The general idea here is to only expire the page when necessary, and we added a new scenario where it is necessary — but only in this case, not necessarily _for all cases_. 

I added this code to the controller, to follow the logic where it currently lives, but in general I think this is best moved to a worker, where it generally, and appropriately, lives in our codebase, i.e. `Comments::BustCacheWorker.rb`.

Reply:
If I'm following you here, then I would suggest this properly belongs in `BustReactableCacheWorker`, as that's more inline with where we currently bust article cache — again, if I am following, we currently call [`EdgeCache::Bust`](https://github.com/forem/forem/blob/main/app/workers/reactions/bust_reactable_cache_worker.rb#L11) on _every_ reaction (because callbacks), but prior to this PR we do _not_ call `EdgeCache::BustArticle`. (It would then work with any/all controllers, even the API, instead of only this one controller.)

If the change we're making here would be to add (or change?) which flavor of `EdgeCache` bust we want, then that's a better place to locate the change, I would think?

Reply:
Yeah @jaw6 actually you're right, we do have `BustReactableCacheWorker`, which doesn't bust the article, but does bust the async request we make, which is otherwise fetched.

I added this to the controller, seeing that we also do some cache busting there (which doesn't seem clean), but this is the wrong place for it. I'm going to update this.

Reply:
@jaw6 I moved this logic into the proper worker.